morpheus/stages/input/arxiv_source.py

```diff
@@ -80,10 +92,11 @@ def _generate_frames(self):
         import arxiv
 
         search_results = arxiv.Search(
-            query="large language models",
+            query=self._query,
             max_results=50,
         )
 
+        # TODO: Move this to a config
```

Comment:
Lets make this a constructor argument `cache_dir`. If its `None` then no caching is performed. If it is set to a value, then we look up the document and load from the cache. On cache miss, if this is set, we save the document

Reply:
If I understand the code correctly, this isn't a cache dir, as `PyPDFLoader` requires a file path (I don't believe it supports a `StringIO`), 

I think what we could/should do is:
* Make a `download_dir` a constructor argument.
* If `None` call `tempfile.mkdtemp` and on complete call `shutil.rmtree`

Reply:
Alterantely, the DFP stages do:
```
cache_dir: str = "./.cache/dfp"
```